internal/client/integration/integration_resource.go

```diff
@@ -0,0 +1,73 @@
+package integrationclient
+
+import (
+	"fmt"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/integration"
+)
+
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+type Client struct {
+	*transport.Client
+}
+
+type ClientService interface {
+	ManageV1alpha1ClusterIntegrationResourceServiceCreate(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationRequest,
+	) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationResponse, error)
+
+	ManageV1alpha1ClusterIntegrationResourceServiceRead(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationFullName,
+	) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationGetIntegrationResponse, error)
+
+	ManageV1alpha1ClusterIntegrationResourceServiceUpdate(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationUpdateIntegrationRequest,
+	) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationUpdateIntegrationResponse, error)
+
+	ManageV1alpha1ClusterIntegrationResourceServiceDelete(
+		*integration.VmwareTanzuManageV1alpha1ClusterIntegrationFullName,
+	) error
+}
+
+const (
+	urlPrefix = "v1alpha1/clusters"
+)
+
+func (c *Client) ManageV1alpha1ClusterIntegrationResourceServiceCreate(
+	request *integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationRequest,
+) (*integration.VmwareTanzuManageV1alpha1ClusterIntegrationCreateIntegrationResponse, error) {
+	requestURL := fmt.Sprintf("%s/%s/integration", urlPrefix, request.Integration.FullName.ClusterName)
```

Comment:
We have a helper function added to contract URLs : helper.ConstructRequestURL
Please use this